server/txSender/dataFormatterDepositTokens.go

```diff
@@ -0,0 +1,72 @@
+package txSender
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+
+	"github.com/multiversx/mx-chain-core-go/data/sovereign"
+	"github.com/multiversx/mx-chain-core-go/hashing"
+)
+
+const (
+	registerBridgeOpsPrefix = "registerBridgeOps"
+	executeBridgeOpsPrefix  = "executeBridgeOps"
+)
+
+type dataFormatterDepositTokens struct {
+	hasher hashing.Hasher
+}
+
+func (df *dataFormatterDepositTokens) createTxsData(bridgeData *sovereign.BridgeOutGoingData) ([][]byte, error) {
+	txsData := make([][]byte, 0)
+	registerBridgeOpData := df.createRegisterBridgeOperationsData(bridgeData)
+	if len(registerBridgeOpData) != 0 {
+		txsData = append(txsData, registerBridgeOpData)
+	}
+
+	return append(txsData, createExecuteBridgeOperationsData(bridgeData.Hash, bridgeData.OutGoingOperations)...), nil
+}
+
+func (df *dataFormatterDepositTokens) createRegisterBridgeOperationsData(bridgeData *sovereign.BridgeOutGoingData) []byte {
+	hashes := make([]byte, 0)
+	hashesHexEncodedArgs := make([]byte, 0)
+	for _, operation := range bridgeData.OutGoingOperations {
+		hashesHexEncodedArgs = append(hashesHexEncodedArgs, "@"+hex.EncodeToString(operation.Hash)...)
+		hashes = append(hashes, operation.Hash...)
+	}
+
+	// unconfirmed operation, should not register it, only resend it
+	computedHashOfHashes := df.hasher.Compute(string(hashes))
+	if !bytes.Equal(bridgeData.Hash, computedHashOfHashes) {
+		return nil
+	}
+
+	registerBridgeOpData := []byte(registerBridgeOpsPrefix +
+		"@" + hex.EncodeToString(bridgeData.AggregatedSignature) +
+		"@" + hex.EncodeToString(bridgeData.Hash) +
+		"@" + hex.EncodeToString(bridgeData.PubKeysBitmap) +
+		"@" + hex.EncodeToString(uint32ToBytes(bridgeData.Epoch)))
```

Comment:
I don't think you need to convert uint32 to bytes

Reply:
It is the same format as it is for every bridge operation (like deposit).
Do you have another suggestion?